File: contests/R1087/B/sol.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

/**
 * C++23 CP Template
 * Compile with: g++ -std=c++23 -O2
 */

void solve() {
    int n;
    cin >> n;

    vector<long long> a(n);

    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    vector<int> ans;

    for (int i = 0; i<n; i++) {
       long long a_i = a[i];

       vector<long long> e;

       int b = 0;

       for (int k = i + 1; k < n; k++) {
           long long a_j = a[k];
           long long mid = a_i + a_j;

           if (a_j < a_i) {
               e.push_back(mid*2);
               b++;
       } else if (a_j > a_i) {
            e.push_back((mid*2)+1);
       }
    }

    sort(e.begin(), e.end());

    int current = b;
    int best = current;

    for (long long x : e) {
        if (x & 1) {
            current++;
        } else {
            current--;
        }
        best = max(best, current);
    }
    ans.push_back(best);
    }

    for (int x : ans) cout << x << ' ';
    cout << '\n';
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int t = 1;
    cin >> t;
    while(t--) solve();
    return 0;
}

File: contests/R1087/B/sol.py
import sys

input = sys.stdin.readline

def main() -> None:
    n = int(input())
    a = []
    best = 0
    
    ans= []

    in_2 = input().split()
    a.extend(map(int, in_2))

    for i in range(n):
        a_i = a[i]
        e = []
        case = 0

        for k in range(i+1, n):
            a_k = a[k]
            mid = a_i + a_k

            if a_k < a_i:
                e.append(mid * 2)
                case += 1
            elif a_k > a_i:
                e.append(mid * 2 + 1)

        e.sort()

        current = case;
        best = current

        for x in e:
            if x & 1:
                current +=1;
            else:
                current -= 1

            best = max(best, current)

        ans.append(best)
    print(*ans)

if __name__ == "__main__":
    t = 1
    t = int(input())
    for _ in range(t):
        main()
    
